pytorch_lightning/core/lightning.py

```diff
@@ -211,9 +214,22 @@ def logger(self):
         """ Reference to the logger object in the Trainer. """
         return self.trainer.logger if self.trainer else None
 
-    def _apply_batch_transfer_handler(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0):
+    def _apply_batch_transfer_handler(
+        self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: Optional[int] = None
+    ) -> Any:
+        device = device or self.device
         batch = self.on_before_batch_transfer(batch, dataloader_idx)
-        batch = self.transfer_batch_to_device(batch, device)
+
+        if is_param_in_hook_signature(self.transfer_batch_to_device, 'dataloader_idx'):
+            batch = self.transfer_batch_to_device(batch, device, dataloader_idx)
+        else:
+            warning_cache.warn(
+                "`transfer_batch_to_device` hook signature has changed in v1.4."
+                " `dataloader_idx` parameter has been added to it. Support for"
+                " the old signature will be removed in v1.6", DeprecationWarning
```

Comment:
will you still show this warning if we are using one dataloader? For one dataloader the dataloader_idx can be optional like it is for our step methods in LightningModule?

Reply:
answered here: https://github.com/PyTorchLightning/pytorch-lightning/pull/6241#discussion_r616598429

Reply:
This suggests that the old signature should still work.  Above, the `dataloader_idx` defaults to `None`, but on the main function in [`core/hooks.py`](https://github.com/PyTorchLightning/pytorch-lightning/pull/6241/files#diff-255f539f021870ad6b64699362bd1b234afe8e606bb4cfd72b54ddba7d0f0326R654), `dataloader_idx` is required (which breaks some of our code that needs to work with earlier versions).